hack/build-local-images.py

```diff
@@ -0,0 +1,136 @@
+		"directory": "images/builder/docker/docker-builder",
+		"binaries": {
+			"openshift": "/usr/bin/openshift"
+		},
+		"files": {}
+	},
+	"openshift/origin-sti-builder": {
+		"directory": "images/builder/docker/sti-builder",
+		"binaries": {
+			"openshift": "/usr/bin/openshift"
+		},
+		"files": {}
+	},
+	"openshift/origin-f5-router": {
+		"directory": "images/router/f5",
+		"binaries": {
+			"openshift": "/usr/bin/openshift"
+		},
+		"files": {}
+	},
+	"openshift/node": {
+		"directory": "images/node",
+		"binaries": {
+			"openshift": "/usr/bin/openshift"
+		},
+		"files": {}
+	},
+	"openshift/openvswitch": {
+		"directory": "images/openvswitch",
+		"binaries": {
+			"openshift": "/usr/bin/openshift"
+		},
+		"files": {}
+	}
+}
+
+def image_rebuild_requested(image):
+	"""
+	An image rebuild is requested if the
+	user provides the image name or image
+	suffix explicitly or does not provide
+	any explicit requests.
+	"""
+	return len(sys.argv) == 1 or (
+		len(sys.argv) > 1 and (
+			image in sys.argv or
+			image.replace("openshift/origin-", '') in sys.argv or
+			image.replace("openshift/", '') in sys.argv
+		)
+	)
+
+def add_to_context(context_dir, source, destination, container_destination):
+	"""
+	Add a file to the context directory
+	and add an entry to the Dockerfile
+	to place it in the container file-
+	sytem at the correct destination.
+	"""
+	debug("Adding file:\n\tfrom {}\n\tto {},\n\tincluding in container at {}".format(source, join(context_dir, destination), container_destination))
+	copyfile(source, join(context_dir, destination))
+	with open(join(context_dir, "Dockerfile"), "a") as dockerfile:
+		dockerfile.write("ADD {} {}\n".format(destination, container_destination))
+
+def debug(message):
+	if getenv("OS_DEBUG"):
+		print "[DEBUG] {}".format(message)
+
+os_root = abspath(join(dirname(__file__),".."))
+os_binpath = join(os_root, "_output","local","bin","linux","amd64")
+
+for image in image_config:
+	if not image_rebuild_requested(image):
+		continue
+
+	print "[INFO] Building {}...".format(image)
+	context_dir = mkdtemp()
+	debug("Created temporary context dir at {}".format(context_dir))
+	mkdir(join(context_dir,"bin"))
+	mkdir(join(context_dir,"src"))
+	with open(join(context_dir, "Dockerfile"), "w+") as dockerfile:
+		dockerfile.write("FROM {}\n".format(image))
+
+	config = image_config[image]
+	for binary in config.get("binaries", []):
+		add_to_context(
+			context_dir,
+			source=join(os_binpath,binary),
+			destination=join("bin",binary),
+			container_destination=config["binaries"][binary]
+		)
+
+	for file in config.get("files", []):
+		add_to_context(
+			context_dir,
+			source=join(os_root,config["directory"],file),
+			destination=join("src",binary),
+			container_destination=config["files"][file]
+		)
+
+	debug("Initiating Docker build with Dockerfile:\n{}".format(open(join(context_dir, "Dockerfile")).read()))
+	call(["docker", "build", "-t", image, "."], cwd=context_dir)
+	rmtree(context_dir)
```

Comment:
you're rebuilding and destroying the context dir for every image?  that means you're going to copy the 300meg openshift binary 8 times if you're building all the images.  This is the kind of thing my bash script was trying to avoid.